syncer/syncer.go

```diff
@@ -716,6 +729,67 @@ func (s *Syncer) addCount(isFinished bool, queueBucket string, tp opType, n int6
 	}
 }
 
+func (s *Syncer) jobsIsEmpty() bool {
+	for idx := range s.jobs {
+		if len(s.jobs[idx]) > 0 {
+			return false
+		}
+	}
+	return true
+}
+
+// updateReplicationLag calc syncer replication lag by job, it called after every batch dml job / one skip job / one ddl
+// job is flushed to target db.
+func (s *Syncer) updateReplicationLag(job *job, queueBucketName string) {
+	if job != nil {
+		failpoint.Inject("BlockSyncerUpdateLag", func(v failpoint.Value) {
+			sv, ok := v.(string)
+			if ok && len(strings.Split(sv, ",")) == 2 {
+				args := strings.Split(sv, ",")
+				jtp := args[0]                // job type
+				t, _ := strconv.Atoi(args[1]) // sleep time
+				if job.tp.String() == jtp {
+					s.tctx.L().Info("BlockSyncerUpdateLag", zap.String("job type", jtp), zap.Int("sleep time", t))
+					time.Sleep(time.Second * time.Duration(t))
+				}
+			} else {
+				s.tctx.L().Info("BlockSyncerUpdateLag failed", zap.Reflect("input v", v))
```

Comment:
I think panic is OK because that's a test. And for error we should use Error level log. And zap.Any is faster than zap.Reflect